### PR TITLE
sdks: add curl vs SDK examples

### PR DESCRIPTION
**Pull request by jacobbednarz**
Introduces a page for mentioning the existence of the SDKs but also how it compares to `curl`.

### CODE REVIEW

File: content/fundamentals/api/sdks.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+pcx_content_type: how-to
+title: SDKs
+weight: 5
+---
+
+# SDKs
+
+Despite many examples using `curl` to demonstrate usage of the Cloudflare API,
```
Comment (irvinebroque) [inline content/fundamentals/api/sdks.md]: Not sure needed? We don't provide the SDKs despite the existence of curl examples :)

File: content/fundamentals/api/sdks.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+your use case and determine which is the best fit.
+
+| Use case                                                    | curl | SDK  |
+| ----------------------------------------------------------- | ---- | ---- |
+| Quick testing within the CLI                                | ✅   | ❌   |
+| Use within bash scripts or CI                               | ✅   | ❌\* |
+| Usage from within an existing application or framework      | ❌   | ✅   |
+| More complex usage where you need to chain together outputs | ❌   | ✅   |
+
+\* You can use SDKs in CI however, it requires installing and additional steps to get working.
```
Comment (irvinebroque) [inline content/fundamentals/api/sdks.md]: Without context this sounds scarier than it is?

**Review by irvinebroque — APPROVED** [review]

File: content/fundamentals/api/sdks.md (hunk 1 above)
Comment (jacobbednarz) [reply content/fundamentals/api/sdks.md]: the thoughts here were to point out that majority of our docs today use curl examples; but the SDKs are an alternative. until we have them alongside one another, i think it's worth mentioning. do you have a wording preference here or perhaps another way to convey that?

File: content/fundamentals/api/sdks.md (hunk 2 above)
Comment (jacobbednarz) [reply content/fundamentals/api/sdks.md]: goal here is to outline the use cases and compare them; open to suggestions to make it less scary if that is the vibe coming off here.

**Comment by jacobbednarz** [comment]
managed to find another page where we mention interacting with Cloudflare so i added the links there too.

**Comment by jacobbednarz** [comment]
@irvinebroque do you think we need both of these? or drop `sdks.md` in favour of `interact-with-cloudflare.md`

File: content/fundamentals/api/reference/sdks.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+There is no definite answer on which you should use. Instead, consider your use case and determine whether curl or an SDK is the best fit.
+
+| Use case                                                    | curl | SDK  |
+| ----------------------------------------------------------- | ---- | ---- |
+| Quick testing within the CLI                                | ✅   | ❌   |
+| Use within bash scripts or CI                               | ✅   | ❌\* |
+| Usage from within an existing application or framework      | ❌   | ✅   |
+| More complex usage where you need to chain together outputs | ❌   | ✅   |
+
+\* You can use SDKs in CI with a few additional steps.
```
Comment (crwaters16) [inline content/fundamentals/api/reference/sdks.md]: @jacobbednarz I rewrote the note but now it seems incredibly vague - were you trying to say you can use them in CI with a few additional (validation/testing) steps?

**Review by crwaters16 — APPROVED** [review]

File: content/fundamentals/api/how-to/make-api-calls.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ For specific guidance on making API calls, refer to the following resources:
 
 *   The product's [Developer Docs section](/products/) for how-to guides.
 *   [API schema docs](/api/) for request and response payloads for each endpoint.
-*   If you are using [golang](https://github.com/cloudflare/cloudflare-go) or [Hashicorp's Terraform](https://github.com/cloudflare/terraform-provider-cloudflare), use our first-party libraries to integrate with Cloudflare's API.
+*   The first-party libraries for[Go](https://github.com/cloudflare/cloudflare-go), [Typescript](https://github.com/cloudflare/cloudflare-typescript), [Python](https://github.com/cloudflare/cloudflare-python), or [Hashicorp's Terraform](https://github.com/cloudflare/terraform-provider-cloudflare).
```
Comment (crwaters16) [inline content/fundamentals/api/how-to/make-api-calls.md]:
```suggestion
*   The first-party libraries for [Go](https://github.com/cloudflare/cloudflare-go), [Typescript](https://github.com/cloudflare/cloudflare-typescript), [Python](https://github.com/cloudflare/cloudflare-python), or [Hashicorp's Terraform](https://github.com/cloudflare/terraform-provider-cloudflare).
```

File: content/fundamentals/api/reference/sdks.md (hunk 3 above)
Comment (crwaters16) [inline content/fundamentals/api/reference/sdks.md]:
```suggestion
\* You can use SDKs in CI with a few additional validation steps.
```

File: content/fundamentals/api/reference/sdks.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+There is no definite answer on which you should use. Instead, consider your use case and determine whether curl or an SDK is the best fit.
+
+| Use case                                                    | curl | SDK  |
+| ----------------------------------------------------------- | ---- | ---- |
+| Quick testing within the CLI                                | ✅   | ❌   |
+| Use within bash scripts or CI                               | ✅   | ❌\* |
+| Usage from within an existing application or framework      | ❌   | ✅   |
+| More complex usage where you need to chain together outputs | ❌   | ✅   |
+
+\* You can use SDKs in CI with a few additional validation steps.
```
Comment (jacobbednarz) [inline content/fundamentals/api/reference/sdks.md]: how about this @crwaters16 ?

```suggestion
\* It is possible, although not straight forward, to use the SDKs within bash scripts or CI environments with additional runtime dependencies and setup.
```

**Pull request merged**